File: Regression_Predictor/Simple Linear Regression/simplelinearregression_nicholas.R
#data preprocessing

#importing the dataset
dataset = read.csv('Salary_Data.csv')
#dataset = dataset[, 2:3]
#splitting the dataset into the Train set and Test set
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#feature scaling
# training_set[, 2:3] = scale(training_set[, 2:3])
# test_set[, 2:3] = scale(test_set[, 2:3])

#fitting simple linear regression to the Training Set
regressor = lm(formula = Salary ~ YearsExperience, 
               data = training_set)

#predicting the Test set results
y_pred = predict(regressor, newdata = test_set)
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set),
            colour = 'green')) +
  ggtitle ('Salary V.S. Experience(Training Set)') +
  xlab('Years of Experience') +
  ylab('Salary') 

   
ggplot() + 
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colour = 'blue') +
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set),
                colour = 'green')) +
  ggtitle ('Salary V.S. Experience(Test Set)') +
  xlab('Years of Experience') +
  ylab('Salary') 

File: Regression_Predictor/Simple Linear Regression/simplelinearregression_nicholas.py
#Simple linear regression

# Preprocess-importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
x = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 1].values

#splitting the dataset into the Training set and Test set
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, test_size = 1/3, random_state = 0)

#feature scaling
"""
from sklearn.preprocessing import StandardScaler
sc_x = StandardScaler()
x_train = sc_x.fit_transform(x_train)
x_test = sc_x.transform(x_test)
"""

#Fitting Simple Linear Regression to the Training Set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(x_train, y_train)

#Predicting the Test set results
y_pred = regressor.predict(x_test)

#visualising the training set results
plt.scatter(x_train, y_train, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue')
plt.title('Salary VS Experience(training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()

#visualising the test set results
plt.scatter(x_test, y_test, color = 'red')
plt.plot(x_train, regressor.predict(x_train), color = 'blue')
plt.title('Salary VS Experience(training set)')
plt.xlabel('Years of Experience')
plt.ylabel('Salary')
plt.show()